tests/Transports.AspNetCore.Tests/Middleware/GetTests.cs

```diff
@@ -65,6 +65,103 @@ public async Task BasicTest()
+    [InlineData("application/json; charset=utf-8", "application/*; charset=utf-8", "application/json; charset=utf-8")]
+    [InlineData("application/json; charset=utf-8", "application/*+json; charset=utf-8", "application/graphql-response+json; charset=utf-8")]
+    [InlineData("application/json; charset=utf-8", "application/pdf", "application/json; charset=utf-8")]
+    [InlineData("application/json; charset=utf-8", "application/json; charset=utf-7", "application/json; charset=utf-8")]
+    [InlineData("application/json", "application/graphql+json", "application/graphql+json; charset=utf-8")]
+    [InlineData("application/json", "application/json", "application/json")]
+    [InlineData("application/json", "application/json; charset=utf-8", "application/json; charset=utf-8")]
+    [InlineData("application/json", "application/json; charset=UTF-8", "application/json; charset=utf-8")]
+    [InlineData("application/json", "APPLICATION/JSON", "application/json")]
+    [InlineData("application/json", "APPLICATION/JSON; CHARSET=\"UTF-8\" ", "application/json; charset=utf-8")]
+    [InlineData("application/json", "*/*; CHARSET=\"UTF-8\" ", "application/json; charset=utf-8")]
+    [InlineData("application/json", "application/*; charset=utf-8", "application/json; charset=utf-8")]
+    [InlineData("application/json", "application/*+json; charset=utf-8", "application/graphql-response+json; charset=utf-8")]
+    [InlineData("application/json", "application/pdf", "application/json")]
+    [InlineData("application/json", "application/json; charset=utf-7", "application/json")]
+    [InlineData(null, "*/*, application/graphql-response+json", "application/graphql-response+json; charset=utf-8")]
+    [InlineData(null, "*/*, application/graphql+json", "application/graphql+json; charset=utf-8")]
+    [InlineData(null, "*/*, application/json", "application/json; charset=utf-8")]
+    [InlineData(null, "*/*, application/pdf", "application/graphql-response+json; charset=utf-8")]
+    [InlineData(null, "application/*, application/graphql-response+json", "application/graphql-response+json; charset=utf-8")]
+    [InlineData(null, "application/*, application/graphql+json", "application/graphql+json; charset=utf-8")]
+    [InlineData(null, "application/*, application/json", "application/json; charset=utf-8")]
+    [InlineData(null, "application/*, application/pdf", "application/graphql-response+json; charset=utf-8")]
+    [InlineData(null, "application/*+json, application/graphql-response+json", "application/graphql-response+json; charset=utf-8")]
+    [InlineData(null, "application/*+json, application/graphql+json", "application/graphql+json; charset=utf-8")]
+    [InlineData(null, "application/*+json, application/json", "application/json; charset=utf-8")]
+    [InlineData(null, "application/*+json, application/pdf", "application/graphql-response+json; charset=utf-8")]
```

Comment:
I think thre should be test for "application/pdf" accept header when server returns error. From https://datatracker.ietf.org/doc/html/rfc7231#section-5.3.2:
![изображение](https://user-images.githubusercontent.com/21261007/191085326-b931dca0-302f-4752-b488-f10b3c84555b.png)

Reply:
According to the draft spec (as of today; it's a moving target!):

> In alignment with the [HTTP 1.1 Accept](https://tools.ietf.org/html/rfc7231#section-5.3.2) specification, when a client does not include at least one supported media type in the Accept HTTP header, the server MUST either:
> 
> - Disregard the Accept header and respond with the default media type of `application/json`, specifying this in the Content-Type header; OR
> - Respond with a 406 Not Acceptable status code and stop processing the request.

I elected the first option.  I would agree that the latter would be better, but I was thinking this change would come in 8.0, as it would involve processing the header earlier in the flow and so would change the design of the middleware.

We could also support additional character encodings; we do so for requests -- why not responses also?  I have no urgency to do so; you're the only one I know of that mentioned using charsets besides utf-8 -- just an idea.

Reply:
agree

Reply:
#932

Reply:
Regarding encodings - spec become rather stric about encoding:
![изображение](https://user-images.githubusercontent.com/21261007/191135085-33959e77-9fb3-480d-a1b6-e1f39f528cb4.png)

MUST is strong requirement )

Reply:
https://github.com/graphql/graphql-over-http/issues/229